lib/job-iteration/enumerator_builder.rb

```diff
@@ -181,6 +181,30 @@ def build_nested_enumerator(enums, cursor:)
       NestedEnumerator.new(enums, cursor: cursor).each
     end
 
+    # Builds Enumerator for while loop iteration.
+    #
+    # @yield Condition to be evaluated before each iteration, the iteration is allowed if a true value is returned.
+    #
+    # @example
+    #   def build_enumerator(params)
+    #     enumerator_builder.while query_model_xyz(params).exists?
+    #   end
+    #
+    #   def each_iteration(count, params)
+    #     query_model_xyz(params).limit(1000).delete_all
+    #   end
+    #
+    #   def query_model_xyz(params)
+    #     Xyz.where(owner_id: params[:owner_id])
+    #   end
+    #
+    def build_while_enumerator(&condition)
+      count = 0
+      Enumerator.new do |yielder|
+        yielder << count += 1 while condition.call
```

Comment:
What would be a use-case for yielding `count`? It will reset to `0` between interruptions and unless there was no interruptions I don't think this number will be useful in any way. 
There should be a way to start counter from the `cursor` position so at least this way `count` will represent the number of times we iterated regardless of whether the job was interrupted or not. 

But perhaps it would be even better to avoid yielding anything until we have a solid use-case for that

Reply:
I agree 100% , I wasn't sure about what to yield to the job. I'll remove the counter.